src/main/java/cat/udl/eps/softarch/tfgfinder/domain/Agree.java

```diff
@@ -0,0 +1,34 @@
+package cat.udl.eps.softarch.tfgfinder.domain;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotBlank;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import org.springframework.format.annotation.DateTimeFormat;
+
+import java.time.ZonedDateTime;
+import java.util.Collection;
+
+@Entity
+@Table(name = "agrees")
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class Agree extends UriEntity<String> {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private String id;
```

Comment:
It is better to use a Long for the identifier instead of a String. That way, because it will be autogenerated, an autoincrement implemented by the underlying database might be re-used. Please, change the String for a Long